### DISCUSSION TITLE
Single tuple causing cursor.executemany ORA-03147: missing mandatory TTC field

### DISCUSSION THREAD

**Discussion by Loncor** [oracle/python-oracledb#29] [General]
I am getting this error when executing the following. Uncommenting the lines below makes it work. So I have concluded if a single tuple in a list is used executemany has a problem. I've got a workaround so not a show stopper. It is not peculiar to this insert as I have tried other tables and differing SQL:-

```
sql_ins_doc_spell = """
            begin
               insert into doc.spell_correct (unknown_word, default_word ) values (:1 , :2);
            exception
                when dup_val_on_index then
                   null;
            end;                              
       """

spell_correct = []

spell_correct.append(('jick', 'jack')) 
#spell_correct.append(('ramana', 'ramona')) 

with connection.cursor() as cursor:
    cursor.executemany(sql_ins_doc_spell, spell_correct)
    connection.commit()
```

**Comment by anthony-tuininga** [comment]
Interesting! I am able to replicate the issue so will look into it further. Thanks for reporting it!

**Reply by cjbj** [reply]
I created issue #30 for this

**Reply by anthony-tuininga** [reply]
A patch is now available for this issue. If you can build yourself you can let us know if it resolves the issue for you!

**Reply by anthony-tuininga** [reply]
And python-oracledb 1.0.2 was just released which contains this patch.
